x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/transform/TransformConfigVersion.java

```diff
@@ -331,10 +331,8 @@ public static Tuple<TransformConfigVersion, TransformConfigVersion> getMinMaxTra
 
     public static TransformConfigVersion getTransformConfigVersionForNode(DiscoveryNode node) {
         String transformConfigVerStr = node.getAttributes().get(TRANSFORM_CONFIG_VERSION_NODE_ATTR);
-        if (transformConfigVerStr != null) {
-            return fromString(transformConfigVerStr);
-        }
-        return fromId(node.getPre811VersionId().orElseThrow(() -> new IllegalStateException("getting legacy version id not possible")));
```

Comment:
Nit: this is the last usage of getPre811VersionId, can we remove it? Or do you want to do that in another PR?

Reply:
I'm removing it in https://github.com/elastic/elasticsearch/pull/115434